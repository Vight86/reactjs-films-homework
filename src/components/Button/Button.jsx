import React from 'react';
import propTypes from 'prop-types';
import styles from './Button.scss';

const Button = ({
  className, onClick, children,
}) => (
  <button
    type="button"
    className={styles[className]}
    onClick={onClick}
    data-text={children}
  >
    {children}
  </button>
);

Button.propTypes = {
  className: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
  children: propTypes.string,
};

Button.defaultProps = {
  children: '',
};

export default Button;
