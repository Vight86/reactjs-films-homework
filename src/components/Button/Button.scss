@import '../../utils/variables';

.button {
  width: 7.14em;
  height: 2.14em;
  border-radius: 0.21em;
  cursor: pointer;
  font-family: $font-primary;
  font-size: 2.8em;
  outline: none;
  transition: 0.3s;
}

.primary {
  @extend .button;

  border: 2px solid currentColor;
  background-color: transparent;
  color: $color-light;

  &:hover {
    border: 2px solid $color-secondary;
    background-color: $color-secondary;
  }
}

.secondary {
  @extend .button;

  border: 2px solid $color-secondary;
  margin-left: 8%;
  background-color: $color-secondary;
  color: $color-light;

  &:hover {
    border: 2px solid $color-light;
    background-color: transparent;
  }
}

.playButton {
  width: 23.4%;
  height: 26.5%;
  border: none;
  margin-bottom: 4%;
  background: url('./playButton.svg') no-repeat center;
  background-color: transparent;
  border-radius: 50%;
  color: transparent;
  cursor: pointer;
  transition: all 0.3s;

  &:hover {
    box-shadow: 0 0 20px 5px $color-secondary;
  }
}

.closeButton {
  position: absolute;
  top: 40px;
  left: 8%;
  padding: 0;
  border: none;
  background: no-repeat;
  color: #fff;
  cursor: pointer;
  font-family: "Times New Roman", serif;
  font-size: 5em;
  line-height: 0.5;
  outline: none;
  transition: all 0.3s;

  &:hover {
    color: $color-secondary;
    transform: rotate(90deg) scale(1.3);
  }
}

.filterButton {
  padding-bottom: 0.4em;
  border: none;
  border-bottom: 4px solid transparent;
  margin-right: 1.8em;
  background: none;
  color: $color-secondary;
  cursor: pointer;
  font-family: $font-primary;
  font-size: 2.8em;
  letter-spacing: 0.04em;
  transition: all 0.2s;

  &:focus {
    border-bottom: 4px solid currentColor;
    font-weight: 700;
    outline: none;
  }
}
