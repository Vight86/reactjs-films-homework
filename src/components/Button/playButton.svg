<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 117 117"><defs><style>.cls-1{fill:#3bb4e0;}.cls-2{fill:#fff;fill-rule:evenodd;}</style></defs><title>Ellipse 3</title><circle class="cls-1" cx="58.5" cy="58.5" r="58.5"/><path class="cls-2" d="M44.56,32.42,82,58.5,44.56,84.58"/></svg>