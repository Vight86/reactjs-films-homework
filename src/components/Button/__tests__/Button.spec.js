import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../index';

let mockOnclick;
let renderer;
let result;

beforeEach(() => {
  mockOnclick = jest.fn();
  renderer = new ShallowRenderer();
  renderer.render(<Button className="primary" onClick={mockOnclick}>test</Button>);
  result = renderer.getRenderOutput();
});

afterEach(() => {
  mockOnclick.mockClear();
  renderer = null;
  result = null;
});

describe('render Button component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });

  it('render onClick event correctly', () => {
    result.props.onClick();
    expect(mockOnclick).toHaveBeenCalled();
  });
});
