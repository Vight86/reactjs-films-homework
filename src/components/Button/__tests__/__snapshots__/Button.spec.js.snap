// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button component render correctly 1`] = `
<button
  className="primary"
  data-text="test"
  onClick={[MockFunction]}
  type="button"
>
  test
</button>
`;
